### PR TITLE
Add more features

### PR DESCRIPTION
**Pull request by g-arjones**
I think this covers the stuff we have discussed. Took a little longer than I expected.

I would like a more thorough review specially on e947fa1aaebdac5d5235814477786c7231438fd9 since (1) I could not test this with the device and (2) had to infer some of the conversions from the code in the IMU's driver without a very deep understanding of them..

I will start working on the component, so please share your opinion about the interfaces you deem adequate. It's been a while since I've done anything with orocos components but if I remember correctly, dynamic properties for the configuration (all the setters in this library) and an output port for the GPS data sound like a good fit IMO. Also, since we are not configuring periodic messages' rates (at least for now), I think I will just discard anything that comes from the device and poll what we want on the update hook of the FileDescriptorActivity. Thoughts?

### CODE REVIEW

**Comment by doudou** [comment]
Looks good generally, will do some testing as soon as you have the component ready.

For the component:
- no dynamic configuration setters unless there's a very good reason (I don't see any in this case)
- avoid having too many small properties. One usually wants to configure all the parameters in one shot, so having a few configuration structures is a better design than a massive list of properties.
- output port for RigidBodyState, gps_base/Solution, gps_base/SatelliteInfo and the internal board state. I'm aware we are losing some info, but that's fine, since the driver reads it we can extract it easily as needed.
- I would very much prefer that you set the PVT message as periodic. The rest could be explicitely polled. The reason is to have the smallest latency when reading the position state. Since GPS is relatively slow, just trigger the "explicit poll" logic for e.g. satellite info after you have read a PVT - so as to not affect the next PVT read too much.

File: src/GPSData.hpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        uint8_t fix_flags;
+        uint8_t additional_flags;
+        uint8_t num_sats;
+        double longitude;  // in degrees
+        double latitude;  // in degrees
+        double height;  // in meters
+        double height_above_mean_sea_level;  // in meters
+        double horizontal_accuracy;  // in meters
+        double vertical_accuracy;  // in meters
+        double vel_north;  // in m/s
```
Comment (doudou) [inline src/GPSData.hpp]: Use `base::Vector3d` for things that go together (e.g. `velocity_ned` here).

File: src/GPSData.hpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        uint32_t time_of_week;  // in milliseconds
+        uint8_t valid;
+        base::Time time;
+        uint32_t time_accuracy;  // in nanoseconds
+        GNSSFixType fix_type;
+        uint8_t fix_flags;
+        uint8_t additional_flags;
+        uint8_t num_sats;
+        double longitude;  // in degrees
```
Comment (doudou) [inline src/GPSData.hpp]: Unfortunately, `gps_base/Solution` is storing latitude and longitude as degrees. Let's not propagate this mess internally, and use radians for all angles. In addition, use `base::Angle` for plain angles (not angular velocities)

File: src/RFInfo.hpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        struct Data {
+            uint8_t block_id;
+            JammingState jamming_state;
+            AntennaStatus antenna_status;
+            AntennaPower antenna_power;
+            uint32_t post_status;
+            uint16_t noise_per_measurement;
+            uint16_t agc_count;
+            uint8_t jamming_indicator;
+            int8_t ofs_i;
```
Comment (doudou) [inline src/RFInfo.hpp:58]: I understand this is opaque, but please document it as the GPS documentation does, that is:

~~~
* The smaller the absolute value of the variable ofsI and ofsQ, the better.
• Ideally, the magnitude of the I-part (magI) and the Q-part (magQ) of the
complex signal should be the same.
~~~

**Comment by doudou** [comment]
I have just implemented the basic oroGen component for the IMU ... might be a source of inspiration as well: https://github.com/tidewise/drivers-orogen-imu_aceinna_openimu/pull/1

**Comment by doudou** [comment]
I do plan to have more than one periodic message there, so will implement that sooner rather than later. So far, the IMU seems to work fine. I have a little problem with velocity covariances, but I'm waiting on the GPS component to do a full log and some statistics.

**Comment by g-arjones** [comment]
Thanks for the review

**Comment by doudou** [comment]
Looks good :+1:

**Review by doudou — APPROVED** [review]
Looks good

**Pull request merged**